File: Backend/info-skjerm-api/Controllers/PostEventsController.cs
/*
This file is for all POST-endpoints for Events
For other endpoints, create new controllers
The route for these endpoints are {baseurl}/PostEvents/{endpoint}

Authored by @Marcus-Aastum
*/

using info_skjerm_api.Model;
using Microsoft.AspNetCore.Mvc;

namespace info_skjerm_api.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class PostEventsController : ControllerBase
    {
        //Defines context for database
        private readonly ApplicationDbContext _context;
        public PostEventsController(ApplicationDbContext context) {
            _context = context;
        }

        //This endpoint creates a new event based on data from body
        [HttpPost("createevent")]
        public IActionResult CreateEvent(Events events)
        {
            //Returns 400 error if event does not have a title
            if(events.title == null || events.title.Length == 0){
                return BadRequest("Event must have a title");
            }

            //Tries to add event, if an error occurs, throw 400
            try{
                _context.Events.Add(events);
                _context.SaveChanges();
                return Ok(events);
            }
            catch (Exception ex) {
                return BadRequest(ex.Message);
            }
        }
        
        
        //DO NOT add GET- PATCH/PUT or DELETE endpoints in this file. Instead, create another conntroller called i.e. "PatchEventsController.cs" 
    }
}

File: Backend/src/controllers/PostEventsController.ts
/*
This file is for all POST-endpoints for Events
For other endpoints, create new controllers
The route for these endpoints are {baseurl}/PostEvents/{endpoint}

Authored by @Marcus-Aastum
*/

import { Router, Request, Response } from "express";
import { context } from "@db";
import { EventsType } from "@models";

const router = Router();

router.post("/createevent", async (req: Request, res: Response) => {
  const eventData = req.body as EventsType;

  if (!eventData.title || eventData.title.length === 0) {
    return res.status(400).send("Event must have a title");
  }

  try {
    const created = await context.events.create(eventData);
    return res.status(200).json(created);
  } catch (err: any) {
    return res.status(400).send(err.message);
  }
});

export default router;